src/core/layout/qgscompositionconverter.cpp

```diff
@@ -1458,10 +1460,18 @@ bool QgsCompositionConverter::readTableXml( QgsLayoutItemAttributeTable *layoutI
         bgAlpha = bgColorElem.attribute( QStringLiteral( "alpha" ) ).toDouble( &alphaOk );
         if ( redOk && greenOk && blueOk && alphaOk )
         {
-          column->mBackgroundColor = QColor( bgRed, bgGreen, bgBlue, bgAlpha );
+          column.mBackgroundColor = QColor( bgRed, bgGreen, bgBlue, bgAlpha );
         }
       }
       layoutItem->mColumns.append( column );
+
+      // sorting columns are now (QGIS 3.14+) handled in a dedicated list
+      Q_NOWARN_DEPRECATED_PUSH
+      for ( const QgsLayoutTableColumn &col : qgis::as_const( layoutItem->mColumns ) )
```

Comment:
this would be easier to read/safer if you enclosed the loop contents in { / }

Reply:
I will switch back to `std::copy_if` then